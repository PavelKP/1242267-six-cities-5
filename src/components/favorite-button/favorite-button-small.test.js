import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import FavoriteButtonSmall from './favorite-button-small';
import mockStoreData from '../../__mocks__/mock-store';

const mockStore = configureStore([]);

describe(`<FavoriteButtonSmall /> render`, () => {

  let store;

  beforeEach(() => {
    store = mockStore(mockStoreData);
  });

  it(`Should FavoriteButton render correctly`, () => {
    const tree = renderer.create(
        <Provider store={store}>
          <FavoriteButtonSmall
            offerId={1}
            buttonClass={`place-card__bookmark-button`}
            iconClass={`place-card__bookmark-icon`}
            activeClass={`place-card__bookmark-button--active`}
          />
        </Provider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
