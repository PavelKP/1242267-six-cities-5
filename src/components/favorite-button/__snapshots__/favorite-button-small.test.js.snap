// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FavoriteButtonSmall /> render Should FavoriteButton render correctly 1`] = `
<button
  className="place-card__bookmark-button button "
  onClick={[Function]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height={19}
    width={18}
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    In bookmarks
  </span>
</button>
`;
